### PR TITLE
Remaster `blobber_test_go`, `Test_flow_reward` 

### PR DESCRIPTION
**Pull request by Sriep**
Waiting on PR https://github.com/0chain/0chain/pull/182 and PR https://github.com/0chain/0chain/pull/186 so it can merge in changes.

[Test_flow_reward ](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/smartcontract/storagesc/blobber_test.go#L129) unit test is bugged. The golang `testing` package can't seem to handle the [testBalances](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/smartcontract/storagesc/balances_test.go#L19) object; In that, it fails to reset the object between `t.Run` calls. Resulting in corrupted data. 

Replaces `Test_flow_reward` with `TestFeesPayments` which tests test payments for some simple cases
* read - Checks service charge, payment to the blobber and increment to read pool
* write - Checks read lock added to challenge pool for later blobber payment and subtracted from locked allocation amount.

This PR handles the fees and rewards testing by lifting the formulae used into a separate [`formulae`](https://github.com/0chain/0chain/blob/830bb341bebb4bd35ddbde7cb9b5513c92d975bd/code/go/0chain.net/smartcontract/storagesc/formulae_test.go) object. Once given various parameters from config files and user input, it produces the theoretical expected result for various actions. Methods added so far:
* amount to charge a write for each unit of time
* amount to charge for a write lock
* blobber reward for a read
* cost of a read

And utility functions
* time remaining in an allocation lock
* convert to time units, must be defined in sc.yaml
* time remaining in allocation lock at the moment of a write

This is related to issue https://github.com/0chain/0chain/issues/104

### CODE REVIEW

**Review by kirillt — APPROVED** [review]
Looks good overall. Please also look at minor comments.

File: code/go/0chain.net/smartcontract/storagesc/formulae_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
+package storagesc
```
[the remark this answers is not in the document]
Comment (Sriep) [reply code/go/0chain.net/smartcontract/storagesc/formulae_test.go:1]: You don't want test files slowing down `go build`. `_test` are automatically omitted from `go build` don't think `test_....go` files are. You could add a `\\ build test` tag but that's complicated. You can have a `testing` subdirectory I think files there are ignored by `go build`.

[the remark this answers is not in the document]
Comment (kirillt) [reply code/go/0chain.net/smartcontract/storagesc/formulae_test.go:1]: Ooh, ok, I thought Go resolves compilation dependencies and skips everything what is not called from main, by default.
`testing` subdirectory sounds good.

File: code/go/0chain.net/smartcontract/minersc/READEME.md [hunk 2]
```diff
@@ -170,20 +113,77 @@ min/max_stake out of this SC boundary.
 #### Interest rate.
 
 Interest rate is % (value in [0; 1) range) that used to calculate interests of
-stake holders. The interests are paid every view change.
+stake holders. The interests payed every view change.
 
 The formula
 ```
-interest_rewards (mint) = stake_capacity * interest_rate
+interest_rewrds (mint) = stake_capacity * interest_rate
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/minersc/READEME.md:120]: Is the `rewrds` a typo ?

**Review by peterlimg — CHANGES_REQUESTED** [review]
I see commits for reverting the changes that added by @kirillt, those changes should have already been reverted on master branch. Would you mind to do a rebase from master, thanks.

**Comment by Sriep** [comment]
Needs reworking due to recent bug fixes, for example https://github.com/0chain/0chain/pull/182 and https://github.com/0chain/0chain/pull/217. Also, overlap with https://github.com/0chain/0chain/pull/174. 

Moved to Issue https://github.com/0chain/0chain/issues/130.

**Pull request closed without merging**